File: mergeTwoLists/mergeTwoLists.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <unordered_map>
#include <stack>

using namespace std;



struct ListNode
{
	int val;
	ListNode *next;
	ListNode() : val(0), next(nullptr) {}
	ListNode(int x) : val(x), next(nullptr) {}
	ListNode(int x, ListNode *next) : val(x), next(next) {}
};


class Solution {
public:
	ListNode* mergeTwoLists(ListNode* list1, ListNode* list2)
	{
		ListNode temp;
		ListNode *head = &temp;

		while (list1 != nullptr && list2 != nullptr)
		{
			if (list1->val < list2->val)
			{
				head->next = list1;
				list1 = list1->next;
			}
			else
			{
				head->next = list2;
				list2 = list2->next;
			}
			head = head->next;
		}
		if (list1 != nullptr)
		{
			head->next = list1;
		}
		if (list2 != nullptr)
		{
			head->next = list2;
		}
		return (temp.next);
	}

	void printLinkedList(ListNode* head)
	{
		ListNode* current = head;
			while (current != nullptr)
			{
				std::cout << current->val << " ";
				current = current->next;
			}
		std::cout << std::endl;
	}
};

int main()
{
	Solution s;
	ListNode* list1 = new ListNode(1);
	list1->next = new ListNode(2);
	list1->next->next = new ListNode(4);

	ListNode* list2 = new ListNode(1);
	list2->next = new ListNode(3);
	list2->next->next = new ListNode(4);

	s.printLinkedList(list1);
	s.printLinkedList(list2);
	ListNode* head = s.mergeTwoLists(list1, list2);
	s.printLinkedList(head);


	return (0);
}

File: mergeTwoLists/mergeTwolists.py

from typing import List, Optional
import collections

class ListNode:
    def __init__(self, val = 0, next = None):
        self.val = val
        self.next = next

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = newHead = ListNode()

        while list1 and list2:
            if list1.val < list2.val:
                newHead.next = list1
                list1 = list1.next
            else:
                newHead.next = list2
                list2 = list2.next
            newHead = newHead.next

        newHead.next = list1 or list2
        return dummy.next




ss = Solution()





